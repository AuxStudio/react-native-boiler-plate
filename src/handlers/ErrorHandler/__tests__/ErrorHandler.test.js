import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import { ErrorHandler } from '..';

jest.mock('../../../scenes/pages/Error'); // Error scene that is displayed on error

const ProblemChild = () => {
  throw new Error('Error thrown from problem child');
  return <div>Error</div>; // eslint-disable-line
};

describe('ErrorHandler', () => {
  const spies = [];
  const dispatch = jest.fn();

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(
        <ErrorHandler dispatch={dispatch}>
          <View />
        </ErrorHandler>,
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle logError', () => {
      const component = renderer.create(
        <ErrorHandler dispatch={dispatch}>
          <View />
        </ErrorHandler>,
      );
      const instance = component.getInstance();
      const error = new Error('foo');

      instance.logError(error);

      expect(dispatch).toHaveBeenCalled();
      expect(dispatch).toMatchSnapshot();
    });
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });

    jest.clearAllMocks();
  });
});
