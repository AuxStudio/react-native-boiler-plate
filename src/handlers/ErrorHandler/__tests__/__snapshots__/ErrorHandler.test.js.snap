// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorHandler methods should handle logError 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "payload": Object {
          "error": [Error: foo],
          "uid": undefined,
        },
        "type": "logError",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`ErrorHandler renders renders the error state 1`] = `null`;

exports[`ErrorHandler renders renders with minimum required props 1`] = `<View />`;
